File: app/src/main/java/eu/napcode/gonoteit/model/note/NoteModel.java
package eu.napcode.gonoteit.model.note;

import eu.napcode.gonoteit.dao.NoteEntity;

public class NoteModel {

    public NoteModel() {
    }

    public NoteModel(NoteEntity noteEntity) {
        this.uuid = noteEntity.getUuid();
        this.title = noteEntity.getTitle();
        this.content = noteEntity.getContent();
        this.id = noteEntity.getId();
        this.imageBase64 = noteEntity.getImageBase64();
    }

    private String uuid;
    private String title;
    private String content;
    private Long id;
    private String imageBase64;

    public String getUuid() {
        return uuid;
    }

    public void setUuid(String uuid) {
        this.uuid = uuid;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getImageBase64() {
        return imageBase64;
    }

    public void setImageBase64(String imageBase64) {
        this.imageBase64 = imageBase64;
    }

    @Override
    public boolean equals(Object obj) {
        return obj instanceof NoteModel &&
                uuid.equals(((NoteModel) obj).uuid) &&
                id.equals(((NoteModel) obj).id) &&
                title.equals(((NoteModel) obj).title) &&
                content.equals(((NoteModel) obj).content) &&
                imageBase64.equals(((NoteModel) obj).imageBase64);
    }
}

File: app/src/main/java/eu/napcode/gonoteit/model/note/NoteModel.kt
package eu.napcode.gonoteit.model.note

import eu.napcode.gonoteit.dao.note.NoteEntity
import eu.napcode.gonoteit.type.Access

open class NoteModel (){
    var title: String? = null
    var content: String? = null
    var id: Long? = null
    var imageBase64: String? = null
    var updatedAt: Long? = null
    var readAccess = Access.INTERNAL
    var writeAccess = Access.INTERNAL
    var date: Long? = null
    var hasAttachment = false

    constructor(noteEntity: NoteEntity): this() {
        title = noteEntity.title
        content = noteEntity.content
        id = noteEntity.id
        imageBase64 = noteEntity.imageBase64
        updatedAt = noteEntity.updatedAt
        readAccess = noteEntity.readAccess
        writeAccess = noteEntity.writeAccess
        date = noteEntity.date
        hasAttachment = noteEntity.hasAttachment
    }

    override fun equals(obj: Any?): Boolean {
        return obj is NoteModel &&
                id == obj.id &&
                areStringsEqual(title, obj.title) &&
                areStringsEqual(content, obj.content) &&
                areStringsEqual(imageBase64, obj.imageBase64) &&
                date == obj.date && hasAttachment == obj.hasAttachment
    }

    private fun areStringsEqual(s: String?, s2: String?): Boolean {
        return s == null && s2 == null || s == s2
    }
}
